### PR TITLE
Add remaining copyright statements, add licenses sanity test

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
This should conclude the endeavor for community.general :-)

CC @gotmax23

CC @russoz and @jirihnidek for https://github.com/ansible-collections/community.general/commit/eb363021263086928011fb1c39d8c37ca4ba4c96

The statements I'm most unsure of are https://github.com/ansible-collections/community.general/commit/7ca7fb087ed7ec5eff88b2d4c1c0d030e0c07bee

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
various

### CODE REVIEW

**Comment by felixfontein** [comment]
Ok, that was a bit too early... I merged the wrong tab. Sorry for that! I won't merge the backport until this - and possibly follow-up PRs to fix what's wrong here - are done.

**Review by gotmax23 — COMMENTED** [review]
I have a couple nitpicks, but I don't think it's a big deal that you merged it as is.

File: tests/integration/targets/jboss/tasks/jboss.yml [hunk 1]
```diff
@@ -4,8 +4,8 @@
 # Integration tests for jboss module.
 # SPDX-License-Identifier: GPL-3.0-or-later
 
-# helloworld.war (got from https://github.com/aeimer/java-example-helloworld-war/) license:
-# MIT License
+# helloworld.war (got from https://github.com/aeimer/java-example-helloworld-war/) is licensed
+# under the MIT license:
```
[the remark this answers is not in the document]
Comment (felixfontein) [reply tests/integration/targets/jboss/tasks/jboss.yml:8]: > So, I'm confused about the licensing situation here. helloworld.war is downloaded as part of the the tests from some S3 bucket but it isn't actually included in this repository?

Yes.

> Also, the MIT license text can be replaced with `# SPDX-License-Identifier: MIT`.

But wouldn't that indicate that parts of **this source file** is licensed under MIT?

File: tests/unit/plugins/modules/system/test_xfconf.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
-# Author: Alexei Znamensky (russoz@gmail.com)
-# Largely adapted from test_redhat_subscription by
-# Jiri Hnidek (jhnidek@redhat.com)
+# Copyright (c) Alexei Znamensky (russoz@gmail.com)
+# Largely adapted from test_redhat_subscription:
+# Copyright (c) Jiri Hnidek (jhnidek@redhat.com)
```
[the remark this answers is not in the document]
Comment (felixfontein) [reply tests/unit/plugins/modules/system/test_xfconf.py:4]: Changed in #5106.

File: tests/unit/plugins/modules/packaging/language/test_cpanm.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
-# Author: Alexei Znamensky (russoz@gmail.com)
-# Largely adapted from test_redhat_subscription by
-# Jiri Hnidek (jhnidek@redhat.com)
+# Copyright (c) Alexei Znamensky (russoz@gmail.com)
+# Largely adapted from test_redhat_subscription:
+# Copyright (c) Jiri Hnidek (jhnidek@redhat.com)
```
[the remark this answers is not in the document]
Comment (felixfontein) [reply tests/unit/plugins/modules/packaging/language/test_cpanm.py:3]: Changed in #5106.

**Comment by felixfontein** [comment]
I'm collecting updates to this PR in #5106.

File: tests/integration/targets/jboss/tasks/jboss.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (gotmax23) [reply tests/integration/targets/jboss/tasks/jboss.yml:8]: > But wouldn't that indicate that parts of this source file is licensed under MIT?

Not necessarily. The copyright and license text will already be picked up by license scanners as applying to that file, anyways.

That section should at least be clarified. Feel free to leave the current MIT text if you'd like.

[the remark this answers is not in the document]
Comment (felixfontein) [reply tests/integration/targets/jboss/tasks/jboss.yml:8]: `reuse` right now prints
```
FileName: ./tests/integration/targets/jboss/tasks/jboss.yml
SPDXID: SPDXRef-21b89fc92d3d59a557caad46c5993b2c
FileChecksum: SHA1: 3fa30c667df64769299e148d83bd8855170a53f6
LicenseConcluded: NOASSERTION
LicenseInfoInFile: GPL-3.0-or-later
FileCopyrightText: <text>Copyright (c) 2017 Alex Eimer
Copyright (c) 2019, Andrew Klychkov (@Andersson007) <xxx></text>
```
So it seems to only identify the GPL-3.0-or-later, but not the MIT license.

In any case, I'd prefer not to depend on external files anyway, probably shipping a small set of text files from which such an application could be built would be a better solution.

[the remark this answers is not in the document]
Comment (gotmax23) [reply tests/integration/targets/jboss/tasks/jboss.yml:8]: > `reuse` right now prints

More generic license scanners will probably pick it up, but I'm fine with leaving it as is.

[the remark this answers is not in the document]
Comment (felixfontein) [reply tests/integration/targets/jboss/tasks/jboss.yml:8]: Ok, then let's leave it like this.

**Pull request merged**